### PR TITLE
Polymer 2.x upgrade

### LINKED ISSUE #37: Polymer 2 upgrade
The sooner, the better.

The dependency`FabricElements/skeleton-carousel` is a Polymer 2 element. It's `hybrid` branch is a hybrid element which does not get maintained ...

also #33 
[WIP]

### PR DESCRIPTION
**Pull request by nomego**
Fixes #37

### CODE REVIEW

**Comment by nomego** [comment]
I have updated @JaySunSyn 's original PR and included our changes since then along with some other fixes.

The main issues left are:
- [ ] Making polymer lint understand that `this._()` and `this.t` are inherited from `Cosmoz.TranslatableBehavior`
- [ ] Testing thoroughly

This should be the next merge to this repo before we do any other work.
Can anyone else in @Neovici/frontend finish this one?

A quick route would be to drop the polymer linting to make sure eslint + tests work and then merge this (with code climate issue) and then open up a new PR to re-enable polymer lint and fix the errors.
As long as the component features are tested.

File: .travis.yml [hunk 1]
```diff
@@ -1,34 +1,39 @@
+---
```
Comment (nomego) [inline .travis.yml]: ?

File: .vscode/settings.json [hunk 2]
```diff
@@ -1,9 +1,15 @@
 
 {
 	"eslint.autoFixOnSave": true,
-	"files.trimTrailingWhitespace": true,
```
Comment (nomego) [inline .vscode/settings.json]: Should still be active

File: cosmoz-image-viewer.html [hunk 3]
```diff
@@ -124,6 +125,7 @@
 				justify-content: center;
 				background-color: black;
 				pointer-events: none;
+				height: 600px;
```
Comment (nomego) [inline cosmoz-image-viewer.html]: Hmm where did this come from? We can keep it if needed but why didn't we need it before?

File: cosmoz-image-viewer.js [hunk 4]
```diff
@@ -1,327 +1,346 @@
...
+				},
+
+				_imageContainerHeight: {
+					type: Number
+				},
+
+				_scroller: {
+					type: Object
+				},
+				t: {
```
Comment (nomego) [inline cosmoz-image-viewer.js]: Should have docs and maybe a fixme

File: cosmoz-image-viewer.js [hunk 5]
```diff
@@ -1,327 +1,346 @@
...
+			};
+		}
+
+		static get observers() {
+			return [
+				'_scrollToPercent(imageLoaded, scrollPercent, _imageContainerHeight)'
+			];
+		}
+
+		_(){
```
Comment (nomego) [inline cosmoz-image-viewer.js]: Docs, fixme, space

File: .travis.yml (hunk 1 above)
Comment (megheaiulian) [reply .travis.yml]: This is for yml linting. A document should start with `---`.

**Pull request merged**